### PR TITLE
Clean incentives state when users unstake all LP

### PR DESCRIPTION
**Pull request by markonmars**

### CODE REVIEW

File: contracts/incentives/tests/tests/test_claim_astro_lp_rewards.rs [hunk 1]
```diff
@@ -638,3 +674,150 @@ fn lp_states_update_correctly() {
...
+    .unwrap();
+
+    set_pending_astro_rewards(
+        &mut deps,
+        lp_denom,
+        mars_incentives_contract,
+        unclaimed_rewards.clone(),
+    );
+
+    // deposit user b
```
Comment (piobab) [inline contracts/incentives/tests/tests/test_claim_astro_lp_rewards.rs:724]: Upper...

File: integration-tests/tests/test_astro_lp_incentives.rs [hunk 2]
```diff
@@ -203,3 +206,115 @@ fn unstake_claims_rewards() {
...
+
+    incentives.unstake_astro_lp(&mut mock_env, &credit_manager, "1".to_string(), lp_coin.clone());
+    mock_env.increment_by_blocks(1);
+    incentives.unstake_astro_lp(&mut mock_env, &credit_manager, "2".to_string(), lp_coin.clone());
+    mock_env.increment_by_blocks(1);
+
+    incentives.stake_astro_lp(&mut mock_env, &credit_manager, "1".to_string(), lp_coin.clone());
+
+    // verify incentives are still 0 - we have not progressed time (blocks) so we should not have rewards
+    let rewards =
```
Comment (piobab) [inline integration-tests/tests/test_astro_lp_incentives.rs:251]: Leaving for a reference..
Check first commit on the branch to see this line fails (it should and it shows the real issue)

**Review by piobab — APPROVED** [review]

**Comment by markonmars** [comment]
Thanks for review! Will prepare migration in separate PR

File: contracts/incentives/src/astro_incentives.rs [hunk 3]
```diff
@@ -232,32 +236,78 @@ fn decrement_staked_lp(
...
+        }
+    } else {
+        ASTRO_USER_LP_DEPOSITS.save(store, key, &new_amount)?;
+    }
 
     Ok(())
 }
 
+fn update_total_staked_lp(store: &mut dyn Storage, lp_coin: &Coin) -> Result<(), ContractError> {
+    let lp_denom = lp_coin.denom.as_str(); // Use &str instead of &String
```
Comment (bobthebuidlr) [inline contracts/incentives/src/astro_incentives.rs:281]: Why this comment? I smell AI assistance... 🤔

File: contracts/incentives/src/astro_incentives.rs [hunk 4]
```diff
@@ -232,32 +236,78 @@ fn decrement_staked_lp(
...
+
+    let total_staked_lp_amount = ASTRO_TOTAL_LP_DEPOSITS
+        .may_load(store, lp_denom)? // Directly use &str
```
Comment (bobthebuidlr) [inline contracts/incentives/src/astro_incentives.rs:284]: Same here

**Review by bobthebuidlr — APPROVED** [review]

File: contracts/incentives/src/astro_incentives.rs (hunk 3 above)
Comment (markonmars) [reply contracts/incentives/src/astro_incentives.rs:281]: HA! goddam copilot

**Pull request merged**